wasm/wasm_source/src/tx_bridge_pool.rs

```diff
@@ -39,26 +38,20 @@ fn apply_tx(ctx: &mut Ctx, signed: Tx) -> TxResult {
             sender,
             &eth_bridge::ADDRESS,
             &nam_addr,
-            None,
             amount.native_denominated(),
             &None,
             &None,
             &None,
         )?;
     } else {
         // Otherwise we escrow ERC20 tokens.
-        let sub_prefix = Some(wrapped_erc20s::sub_prefix(&asset));
-        let amount = amount.denominated(
-            &eth_bridge::ADDRESS,
-            sub_prefix.as_ref(),
-            ctx,
-        )?;
+        let token = wrapped_erc20s::token(&asset);
+        let amount = amount.denominated(&eth_bridge::ADDRESS, ctx)?;
```

Comment:
do we actually need this amount to be denominated? the only consumer of the denomination in `tx_prelude/src/token.rs` is the masp, for shielded txs

Reply:
No, it's not needed. But `token::transfer` requires `DenominatedAmount` now.